rel/files/mongooseim.toml

```diff
@@ -0,0 +1,362 @@
+    host = "_"
+    path = "/api/rooms/:id/users/[:user]"
+
+  [[listen.cowboy.modules.mongoose_client_api_rooms_messages]]
+    host = "_"
+    path = "/api/rooms/[:id]/messages"
+
+  [[listen.cowboy.modules.cowboy_swagger_redirect_handler]]
+    host = "_"
+    path = "/api-docs"
+
+  [[listen.cowboy.modules.cowboy_swagger_json_handler]]
+    host = "_"
+    path = "/api-docs/swagger.json"
+
+  [[listen.cowboy.modules.cowboy_static]]
+    host = "_"
+    path = "/api-docs/[...]"
+    type = "priv_dir"
+    app = "cowboy_swagger"
+    content_path = "swagger"
+
+[[listen.cowboy]]
+  port = 5288
+  transport.num_acceptors = 10
+  transport.max_connections = 1024
+
+  [[listen.cowboy.modules.mongoose_api]]
+    host = "localhost"
+    path = "/api"
+    handlers = ["mongoose_api_metrics", "mongoose_api_users"]
+
+[[listen.c2s]]
+  port = 5222
+  certfile = "priv/ssl/fake_server.pem"
+  starttls = true
+  zlib = 10000
+  access = "c2s"
+  shaper = "c2s_shaper"
+  max_stanza_size = 65536
+  dhfile = "priv/ssl/fake_dh_server.pem"
+
+[[listen.c2s]]
+  port = 5223
+  zlib = 4096
+  access = "c2s"
+  shaper = "c2s_shaper"
+  max_stanza_size = 65536
+
+[[listen.s2s]]
+  port = 5269
+  shaper = "c2s_shaper"
+  max_stanza_size = 131072
+  dhfile = "priv/ssl/fake_dh_server.pem"
+
+[[listen.service]]
+  port = 8888
+  access = "all"
+  shaper_rule = "fast"
+  ip_address = "127.0.0.1"
+  password = "secret"
+
+[[listen.service]]
+  port = 8666
+  access = "all"
+  conflict_behaviour = "kick_old"
+  shaper_rule = "fast"
+  ip_address = "127.0.0.1"
+  password = "secret"
+
+[[listen.service]]
+  port = 8189
+  access = "all"
+  hidden_components = true
+  shaper_rule = "fast"
+  ip_address = "127.0.0.1"
+  password = "secret"
+
+[s2s]
+  use_starttls = "optional"
+  certfile = "priv/ssl/fake_server.pem"
+  default_policy = "allow"
+  outgoing_port = 5299
+
+  [[s2s.address]]
+    host = "fed1"
+    ip_address = "127.0.0.1"
+
+[auth]
+  methods = ["rdbms"]
+  password.format = "scram"
+  password.hash = ["sha256"]
+  scram_iterations = 64
+
+[[host_config]]
+  host = "anonymous.localhost"
+
+  [host_config.auth]
+    methods = ["anonymous"]
+    allow_multiple_connections = true
+    anonymous_protocol = "both"
+
+[outgoing_pools.redis.global_distrib]
+  scope = "single_host"
+  host = "localhost"
+  workers = 10
+
+[outgoing_pools.rdbms.default]
+  scope = "global"
+  workers = 5
+
+  [outgoing_pools.rdbms.default.connection]
+    driver = "pgsql"
+    host = "localhost"
+    database = "ejabberd"
+    username = "ejabberd"
+    password = "mongooseim_secret"
+    tls.required = true
+    tls.verify_peer = true
+    tls.cacertfile = "priv/ssl/cacert.pem"
+    tls.server_name_indication = false
+
+[shaper.normal]
+  max_rate = 1000
+
+[shaper.fast]
+  max_rate = 50_000
+
+[shaper.mam_shaper]
+  max_rate = 1
+
+[shaper.mam_global_shaper]
+  max_rate = 1000
+
+[acl.local]
+  user_regexp = ""
+
+[access]
+  max_user_sessions = [
+    {acl = "all", value = 10}
+  ]
```

Comment:
Why is this an array and not a table like the rest of the config? It looks like we deal with key-value pairs here as well

Reply:
These are lists of rules and the order of rules matters, so they cannot be KV pairs. The alternative syntax would be:

```
[[access.max_user_offline_messages]]
  acl = admin
  value = 5000


[[access.max_user_offline_messages]]
  acl = all
  value = 100
```

For me it is less readable.